promql/engine_test.go

```diff
@@ -47,9 +47,10 @@ func TestMain(m *testing.M) {
 func TestQueryConcurrency(t *testing.T) {
 	maxConcurrency := 10
 
-	dir, err := os.MkdirTemp("", "test_concurrency")
-	require.NoError(t, err)
-	defer os.RemoveAll(dir)
+	dir := t.TempDir()
+	t.Cleanup(func() {
```

Comment:
t.TempDir() TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete. 
```suggestion
```

Reply:
@roidelapluie thank you for the explanation, all make sense, I will update pr